src/server/stream_family.cc

```diff
@@ -2297,6 +2310,18 @@ void XReadImpl(CmdArgList args, std::optional<ReadOpts> opts, ConnectionContext*
       if (streamCompareID(&last_id, &requested_sitem.id.val) >= 0) {
         block = false;
       }
+    } else {
+      if (opts->read_group) {
+        // if the key is not found,
+        // return NoGroupOrKey error.
+        // We are simply mimicking Redis' error message here...
+        // However, we could actually report more precise error message...
+        string error_msg =
+            NoGroupOrKey(stream, opts->group_name, " in XREADGROUP with GROUP option");
+        cntx->transaction->Conclude();
+        (*cntx)->SendError(error_msg);
```

Comment:
you do not need to use `(*cntx)` anymore when replying an error. `cntx->SendError` works too.

Reply:
thanks! also changed all the other places in `stream_family.cc`